app/src/main/java/com/bogdan/codeforceswatcher/features/actions/ActionActivity.kt

```diff
@@ -36,15 +37,16 @@ class ActionActivity : AppCompatActivity() {
     }
 
     private fun initData() {
-        val cfAction = intent.getSerializableExtra(ACTION_ID) as CFAction
+        val cfAction = store.state.actions.actions.find { it.id == intent.getIntExtra(ACTION_ID, -1) }
+                ?: throw IllegalStateException()
 
         pageTitle = cfAction.blogEntry.title
         link = buildPageLink(cfAction)
     }
 
     private fun buildPageLink(cfAction: CFAction) =
```

Comment:
You can write it as `private fun buildPageLink(cfAction: CFAction) = cfAction.comment?.let { commentId -> } ?: getString(R.string.blog_entry_url, cfAction.blogEntry.id)`.